dsl/interpreter.py

```diff
@@ -0,0 +1,164 @@
+import os
+import re
+import logging
+
+from dsl.parser import Grammar
+
+logger = logging.getLogger('example_logger')
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
+DSL_EXTENSION = '.str'
+MODULES_DIR_NAME = 'dsl/modules'
+modules = {}
+def load_modules():
+    content = os.listdir(MODULES_DIR_NAME)
+    for file in content:
+        if os.path.isfile(os.path.join(MODULES_DIR_NAME, file)) and file.endswith('.py'):
+            name = file[:len(file)-3]
+            modules[name] = exec()
+
+
+def parse_file(input_name: str) -> str | None:
+    if input_name.endswith('.py'):
+        return input_name
+    if not input_name.endswith(DSL_EXTENSION):
+        logger.info(f'File {input_name} has incorrect extension')
+        return None
+    if not os.path.isfile(input_name):
+        logger.error(f'File {input_name} does not exist')
+    file = open(input_name, "r")
+    lines = file.readlines()
+    # print whole file to a string with removing comments
+    is_commented = False
+    data = ''
+    # gr = Grammar()
+    # clean from comments
+    for line in lines:
+        line = line.strip()
+        ind_line_commented = len(line)
+        for i, c in enumerate(line):
+            if c == '@':
+                is_commented = not is_commented
+            if c == '#' and not is_commented and ind_line_commented == len(line):
+                ind_line_commented = i
+            if not (is_commented or ind_line_commented != len(line)):
+                data += c
+    # code = gr.parse(data)
+    code = simple_transform(data)
+    new_file_name = input_name[:len(input_name) - len(DSL_EXTENSION)] + '.py'
+    with open(new_file_name, 'w') as f:
+        f.write(code)
+    # close file
+    file.close()
+    return new_file_name
+
+def simple_transform(code='RETURN 0'):
+    python_code = 'import numpy as np\n'
+    code = code.replace('\n', ' \n ')
+    code = code.replace('{', ' {')
+    code = code.replace('}', ' }')
+    tokens = code.split(' ')
+    # print(code)
+    # print(tokens)
+    indent = 0
+    prefix = 0
+    if_flag = 0
+    else_flag = 0
+    for_flag = 0
```

Comment:
These should be `bool`: there cannot be 2 tokens, because `{` will decrease the token number and 2 such tokens cannot be together without `{`